src/System.IO.Packaging/src/System/IO/Packaging/PackUriHelper.cs

```diff
@@ -0,0 +1,1306 @@
+                //Case Fold the partUri string to Invariant Upper case (this helps us perform case insensitive comparison)
+                //We follow the Simple case folding specified in the Unicode standard
+
+                if (_isNormalized)
+                    return _partUriString;
+                else
+                    return _partUriString.ToUpperInvariant();
+            }
+
+            private ValidatedPartUri GetNormalizedPartUri()
+            {
+                if (IsNormalized)
+                    return this;
+                else
+                    return new ValidatedPartUri(_normalizedPartUriString,
+                                                true /*isNormalized*/,
+                                                false /*computeIsRelationship*/,
+                                                IsRelationshipPartUri);
+            }
+
+            private int Compare(ValidatedPartUri otherPartUri)
+            {
+                //If otherPartUri is null then we return 1
+                if (otherPartUri == null)
+                    return 1;
+
+                //Compare the normalized uri strings for the two part uris.
+                return String.CompareOrdinal(this.NormalizedPartUriString, otherPartUri.NormalizedPartUriString);
+            }
+
+            //------------------------------------------------------
+            //
+            //  Private Members
+            //
+            //------------------------------------------------------
+
+            private ValidatedPartUri _normalizedPartUri;
+            private string _partUriString;
+            private string _normalizedPartUriString;
+            private string _partUriExtension;
+            private bool _isNormalized;
+            private bool _isRelationshipPartUri;
+
+            //String Uppercase variants
+
+            private static readonly string s_relationshipPartUpperCaseExtension = ".RELS";
+            private static readonly string s_relationshipPartUpperCaseSegmentName = "_RELS";
+            private static readonly string s_relsrelsUpperCaseExtension = String.Concat(s_relationshipPartUpperCaseExtension, s_relationshipPartUpperCaseExtension);
+
+            //need to use the private constructor to initialize this particular partUri as we need this in the 
+            //IsRelationshipPartUri, that is called from the constructor.
+            private static readonly Uri s_containerRelationshipNormalizedPartUri = new ValidatedPartUri("/_RELS/.RELS",
+                                                                                                         true /*isnormalized*/,
+                                                                                                         false /*computeIsRelationship*/,
+                                                                                                         true /*IsRelationship*/);
+
+            private static readonly char[] s_forwardSlashSeparator = { '/' };
```

Comment:
I think I saw at least one other place where the same thing was declared.  Should they be consolidated?